launch/launch/actions/execute_process.py

```diff
@@ -436,12 +443,15 @@ def __expand_substitutions(self, context):
 
         returncode = await self._subprocess_protocol.complete
         if returncode == 0:
-            _logger.info('process[{}]: process has finished cleanly'.format(name, pid))
+            _logger.info('process[{}]: process has finished cleanly'.format(name))
         else:
             _logger.error("process[{}] process has died [pid {}, exit code {}, cmd '{}'].".format(
                 name, pid, returncode, ' '.join(cmd)
             ))
         await context.emit_event(ProcessExited(returncode=returncode, **process_event_args))
+        if self.__required:
+            _logger.info('process[{}] was required: shutting down launched system'.format(name))
+            await context.emit_event(Shutdown(reason="Required process {!r} exited".format(name)))
```

Comment:
It will? This file is littered with both single and double quotes. I don't have a problem changing, but if that's a linter issue, should they _all_ be fixed?

Reply:
@kyrofa Please run the unit tests of the package you are modifying locally.

Basically single quotes are used as long as that doesn't resulting in needing to escape single quotes within the string. Otherwise double quotes are used.

Reply:
Ah, thank you for the developer guide. Might I suggest linking to it from [here](https://github.com/ros2/launch/blob/master/CONTRIBUTING.md)?

Reply:
Don't worry @dirk-thomas, I definitely ran the tests. Just didn't realize everything needed to be on master instead of following [this](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#python) (including ament_lint). I've fixed that, and now I see this error (and other unrelated ones went away, nice).

I've pushed the fix.

Reply:
Usually we single quote expanded variables in output (to easily detect cases where it's unexpectedly an empty string), so my I suggest:

```python
"Required process '{!r}' exited with exit code '{}'.".format(name, returncode)
```

Reply:
@wjwwood that's actually what the `!r` does in this case:

```
>>> name = 'foo'
>>> print('name: {!r}'.format(name))
name: 'foo'
```

Would you prefer my removing that and adding the quotes explicitly? I can certainly add the return code.

Reply:
Oh, I see, I didn't think about `!r` doing that. It's up to you.